### PR TITLE
TST/COMPAT: update csv test to infer time with pyarrow>=6.0

### LINKED ISSUE #44231: BUG: test_date_col_as_index_col[pyarrow] failing locally with pyarrow 6.0.0
### 

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the master branch of pandas.

### Reproducible Example

```python
>   raise_assert_detail(obj, msg, lobj, robj, index_values=index_values)
E   AssertionError: DataFrame.iloc[:, 1] (column name="X2") are different
E   
E   DataFrame.iloc[:, 1] (column name="X2") values are different (100.0 %)
E   [index]: [1999-01-27T19:00:00.000000000, 1999-01-27T20:00:00.000000000, 1999-01-27T21:00:00.000000000, 1999-01-27T21:00:00.000000000, 1999-01-27T22:00:00.000000000]
E   [left]:  [18:56:00, 19:56:00, 20:56:00, 21:18:00, 21:56:00]
E   [right]: [ 18:56:00,  19:56:00,  20:56:00,  21:18:00,  21:56:00]
```

### Issue Description

cc @jorisvandenbossche 

### Expected Behavior

NA

### Installed Versions

<details>

Replace this line with the output of pd.show_versions()

</details>

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Closes #44231

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
Thanks @jorisvandenbossche. This has now starting failing on master and so I guess probably 1.3.x too. Will milestone 1.3.5 for backport.

File: pandas/compat/pyarrow.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
     pa_version_under3p0 = _palv < Version("3.0.0")
     pa_version_under4p0 = _palv < Version("4.0.0")
     pa_version_under5p0 = _palv < Version("5.0.0")
+    pa_version_under6p0 = _palv < Version("6.0.0")
```
Comment (simonjayhawkins) [inline pandas/compat/pyarrow.py:15]: I think also need to add `pa_version_under6p0 = True` below

Comment (jorisvandenbossche) [reply pandas/compat/pyarrow.py:15]: Ah, yes, thanks

**Comment by lithomas1** [comment]
> Thanks @jorisvandenbossche. This has now starting failing on master and so I guess probably 1.3.x too. Will milestone 1.3.5 for backport.

The new pyarrow engine should only be present on master, so I guess backport is not needed.

**Comment by jorisvandenbossche** [comment]
Good point.

I am not aware of any discussion about an option to turn it off (looking at https://issues.apache.org/jira/browse/ARROW-11247 now)

**Comment by jorisvandenbossche** [comment]
The sdist 3.10 build was failing (unrelated I suppose), is that a know failure ? (I don't directly see an issue about it)

**Pull request merged**